src/main/java/io/supertokens/session/accessToken/AccessToken.java

```diff
@@ -208,11 +283,93 @@ public static class AccessTokenInfo {
             this.userData = userData;
             this.antiCsrfToken = antiCsrfToken;
             this.timeCreated = timeCreated;
-            this.lmrt = lmrt;
+            this.version = version;
+        }
+
+        static AccessTokenInfo fromJSON(JsonObject payload, VERSION version) throws TryRefreshTokenException {
+            JsonElement parentRefreshTokenHash = payload.get("parentRefreshTokenHash1");
+            JsonElement antiCsrfToken = payload.get("antiCsrfToken");
+
+            if (version == VERSION.V3) {
+                checkRequiredPropsExist(payload, requiredPropsV3);
+                JsonObject userData = new JsonObject();
+
+                for (Map.Entry<String, JsonElement> element : payload.entrySet()) {
+                    if(Arrays.stream(protectedPropNames).noneMatch(element.getKey()::equals)){
+                        userData.add(element.getKey(), element.getValue());
+                    }
+                }
+
+                return new AccessTokenInfo(
+                        payload.get("sessionHandle").getAsString(),
+                        payload.get("sub").getAsString(),
+                        payload.get("refreshTokenHash1").getAsString(),
+                        payload.get("exp").getAsLong() * 1000,
+                        parentRefreshTokenHash.isJsonNull() ? null : parentRefreshTokenHash.getAsString(),
+                        userData,
+                        antiCsrfToken.isJsonNull() ? null : antiCsrfToken.getAsString(),
+                        payload.get("iat").getAsLong() * 1000,
+                        version
+                );
+            } else {
+                checkRequiredPropsExist(payload, requiredPropsV2);
+                return new AccessTokenInfo(
+                        payload.get("sessionHandle").getAsString(),
+                        payload.get("userId").getAsString(),
+                        payload.get("refreshTokenHash1").getAsString(),
+                        payload.get("expiryTime").getAsLong(),
+                        parentRefreshTokenHash.isJsonNull() ? null : parentRefreshTokenHash.getAsString(),
+                        payload.get("userData").getAsJsonObject(),
+                        antiCsrfToken.isJsonNull() ? null : antiCsrfToken.getAsString(),
+                        payload.get("timeCreated").getAsLong(),
+                        version
+                );
+            }
+
+        }
+
+        JsonObject toJSON() throws UnauthorisedException {
+            JsonObject res = new JsonObject();
+            if (this.version == VERSION.V3) {
+                res.addProperty("sub", this.userId);
+                res.addProperty("exp", this.expiryTime / 1000);
+                res.addProperty("iat", this.timeCreated / 1000);
```

Comment:
this will drop the decimal since we are doing long / int - is this expected?

Reply:
it is, `iat` and `exp` are integers values that contain seconds not milliseconds.